### ISSUE TITLE
Detect `slots=True`-specific issues in dataclasses

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#3094]
The conformance tests require us to detect certain issues related to `slots=True` in dataclasses. For example:

```py
# This should generate an error because __slots__ is already defined.
@dataclass(slots=True)  # E[DC1]
class DC1:  # E[DC1]
    x: int

    __slots__ = ()
```

See [these tests](https://shark.fish/typing-conformance-report/dataclasses_slots.html) for details.
